pkg/model/azuremodel/network.go

```diff
@@ -122,7 +124,32 @@ func (b *NetworkModelBuilder) Build(c *fi.CloudupModelBuilderContext) error {
 			SourceAddressPrefixes:    &k8sAccessIPv6,
 			SourcePortRange:          fi.PtrTo("*"),
 			DestinationAddressPrefix: fi.PtrTo("*"),
-			DestinationPortRange:     fi.PtrTo("443"),
+			DestinationPortRange:     fi.PtrTo(strconv.Itoa(wellknownports.KubeAPIServer)),
+		})
+	}
+	if b.Cluster.UsesNoneDNS() {
+		// TODO: Limit access to necessary source address prefixes instead of "0.0.0.0/0" and "::/0"
```

Comment:
It would be better to limit this out of the gate. At the bare minimum, it should be limited to `networkCIDR`.

Reply:
It can't be limited, because you have internal IPs (with private address) and external LB (with public address).
Whitelisting the networkCIDR does nothing.